f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py

```diff
@@ -1230,28 +1225,50 @@ def _common_service_handler(self, service, delete_partition=False):
                      'do_not_delete_subnets': []}
 
             LOG.debug("XXXXXXXXX: Pre assure service")
+            # pdb.set_trace()
             self.lbaas_builder.assure_service(service,
                                               traffic_group,
                                               all_subnet_hints)
             LOG.debug("XXXXXXXXX: Post assure service")
 
             if self.network_builder:
                 start_time = time()
-                self.network_builder.post_service_networking(
-                    service, all_subnet_hints)
-                LOG.debug("    _post_service_networking took %.5f secs" %
-                          (time() - start_time))
+                try:
+                    self.network_builder.post_service_networking(
+                        service, all_subnet_hints)
+                except Exception as error:
+                    LOG.error("Post-network exception: icontrol_driver: %s",
+                              error.message)
+
+                    if lb_provisioning_status != plugin_const.PENDING_DELETE:
+                        loadbalancer['provisioning_status'] = \
+                            plugin_const.ERROR
+                        raise error
+
+                if time() - start_time > .001:
+                    LOG.debug("    _post_service_networking "
+                              "took %.5f secs" % (time() - start_time))
 
+        except f5ex.NetworkNotReady as error:
+            pass
+        except Exception as err:
+            LOG.exception(err)
+        finally:
             # only delete partition if loadbalancer is being deleted
-            if delete_partition:
+            if lb_provisioning_status == plugin_const.PENDING_DELETE:
```

Comment:
No, we don't do that anymore.  We used to use the delete_folder_contents method, which did blow everything away, but that is no longer the case.  In this case we try to delete the folder, but we fail because the folder contains the other loadbalancer(s).  In the tenant we handle the case where a folder deletion event occurs.